test/e2e/network/networking_perf.go

```diff
@@ -160,3 +172,227 @@ var _ = SIGDescribe("Networking IPerf [Experimental] [Slow] [Feature:Networking-
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: v1.PodSpec{
+					TerminationGracePeriodSeconds: &zero,
+					Containers: []v1.Container{
+						{
+							Name:    "iperf2-server",
+							Image:   imageutils.GetE2EImage(imageutils.Agnhost),
+							Command: []string{"iperf"},
+							Args: []string{
+								"-s",
+								"-p",
+								"6789",
+							},
+							Ports: []v1.ContainerPort{
+								{
+									ContainerPort: 6789,
+									Protocol:      v1.ProtocolTCP,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	deployment, err := client.AppsV1().Deployments(namespace).Create(context.TODO(), deploymentSpec, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("deployment %q Create API error: %v", deploymentSpec.Name, err)
+	}
+	framework.Logf("Waiting deployment %q to complete", deploymentSpec.Name)
+	err = e2edeployment.WaitForDeploymentComplete(client, deployment)
+	if err != nil {
+		return nil, fmt.Errorf("deployment %q failed to complete: %v", deploymentSpec.Name, err)
+	}
+	return deployment, nil
+}
+
+func iperf2ServerService(client clientset.Interface, namespace string) (*v1.Service, error) {
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: "iperf2-server"},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{
+				"app": "iperf2-server",
+			},
+			Ports: []v1.ServicePort{
+				{Protocol: v1.ProtocolTCP, Port: 6789},
+			},
+		},
+	}
+	return client.CoreV1().Services(namespace).Create(context.TODO(), service, metav1.CreateOptions{})
+}
+
+func iperf2ClientDaemonSet(client clientset.Interface, namespace string) (*appsv1.DaemonSet, error) {
+	zero := int64(1)
+	labels := map[string]string{"app": "iperf2-client"}
+	spec := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "iperf2-clients",
+			Labels: labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: v1.PodSpec{
+					Tolerations: []v1.Toleration{
+						{
+							Key:      "node-role.kubernetes.io/master",
+							Operator: v1.TolerationOpEqual,
+							Effect:   v1.TaintEffectNoSchedule,
+						},
+					},
+					Containers: []v1.Container{
+						{
+							Name:    "iperf2-client",
+							Image:   imageutils.GetE2EImage(imageutils.Agnhost),
+							Command: []string{"/bin/sh"},
+							Args:    []string{"-c", "sleep 99999999"},
+						},
+					},
+					TerminationGracePeriodSeconds: &zero,
+				},
+			},
+		},
+		Status: appsv1.DaemonSetStatus{},
+	}
+
+	ds, err := client.AppsV1().DaemonSets(namespace).Create(context.TODO(), spec, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("daemonset %s Create API error: %v", spec.Name, err)
+	}
+	return ds, nil
+}
+
```

Comment:
comments on these scripts and maybe move the `IPerfResult` to colocate w / them ?

Reply:
👍